frontend/src/components/tracing/tracing.tsx

```diff
@@ -0,0 +1,325 @@
+        id={hiddenInputElementId}
+        defaultValue={hoveredCellId || ""}
+        hidden={true}
+        ref={hiddenInputRef}
+      />
+      {run.cellRuns.map((cellRun) => (
+        <TraceRow
+          key={cellRun.cellId}
+          cellRun={cellRun}
+          hovered={cellRun.cellId === hoveredCellId}
+          dispatchHoverEvent={dispatchHoverEvent}
+        />
+      ))}
+    </div>
+  );
+
+  if (chartPosition === "above") {
+    return (
+      <div key={run.runId} className="flex flex-col">
+        <pre className="font-mono font-semibold">
+          {TraceTitle}
+          {!collapsed && (
+            <Chart
+              vegaSpec={vegaSpec}
+              width={320}
+              height={120}
+              signalListeners={handleVegaSignal}
+            />
+          )}
+          {!collapsed && TraceRows}
+        </pre>
+      </div>
+    );
+  }
+
+  return (
+    <div key={run.runId} className="flex flex-row">
+      <pre className="font-mono font-semibold">
+        {TraceTitle}
+        {!collapsed && TraceRows}
+      </pre>
+      {!collapsed && (
+        <Chart
+          className="-mt-0.5"
+          vegaSpec={vegaSpec}
+          width={240}
+          height={100}
+          signalListeners={handleVegaSignal}
+        />
+      )}
+    </div>
+  );
+};
+
+interface TraceRowProps {
+  cellRun: CellRun;
+  hovered: boolean;
+  dispatchHoverEvent: (cellId: CellId | null) => void;
+}
+
+const TraceRow: React.FC<TraceRowProps> = ({
+  cellRun,
+  hovered,
+  dispatchHoverEvent,
+}: TraceRowProps) => {
+  const elapsedTimeStr = cellRun.elapsedTime
+    ? formatElapsedTime(cellRun.elapsedTime * 1000)
+    : "-";
+
+  const elapsedTimeTooltip = cellRun.elapsedTime ? (
+    <span>
+      This cell took <ElapsedTime elapsedTime={elapsedTimeStr} /> to run
+    </span>
+  ) : (
+    <span>This cell has not been run</span>
+  );
+
+  const handleMouseEnter = () => {
+    dispatchHoverEvent(cellRun.cellId);
+  };
+
+  const handleMouseLeave = () => {
+    dispatchHoverEvent(null);
+  };
+
+  return (
+    <div
+      className={cn(
+        "flex flex-row gap-2 py-1 px-1 opacity-70 hover:bg-[var(--gray-3)] hover:opacity-100",
+        hovered && "bg-[var(--gray-3)] opacity-100",
+      )}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      <span className="text-[var(--gray-10)]">
+        [{formatLogTimestamp(cellRun.startTime)}]
+      </span>
+      <span className="text-[var(--gray-10)]">
+        (<CellLink cellId={cellRun.cellId} />)
+      </span>
+      <span className="w-40 truncate">{cellRun.code}</span>
+
+      <div className="flex flex-row gap-1 basis-12 justify-end">
+        <Tooltip content={elapsedTimeTooltip}>
+          <span className="text-[var(--gray-10)]">{elapsedTimeStr}</span>
+        </Tooltip>
+
+        {/* TODO: Shouldn't use favicon and need to add an image for 'queued' state. */}
+        <Tooltip content={cellRun.status}>
+          <img
```

Comment:
if you don't get one, it could be ok to use colors and lucid icons:

Queued: https://lucide.dev/icons/loader (gray)
Running: https://lucide.dev/icons/loader-circle + `animate-spine` (darker-gray)
Done: https://lucide.dev/icons/circle-check (green)
Failed: https://lucide.dev/icons/circle-x (red)

Reply:
I'm opting to use ellipsis for queued to maintain design consistency with the editor. But let me know if you prefer a loader.